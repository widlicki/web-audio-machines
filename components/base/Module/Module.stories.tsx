import { ComponentMeta, ComponentStory } from '@storybook/react'
import Module from './Module'

export default {
  title: 'components/base/Module',
  component: Module,
} as ComponentMeta<typeof Module>

const Template: ComponentStory<typeof Module> = (args) => (
  <Module {...args}>
    <h4>Module Content</h4>
  </Module>
)

export const DefaultModule = Template.bind({})
DefaultModule.args = {
  title: 'Module Title',
}
