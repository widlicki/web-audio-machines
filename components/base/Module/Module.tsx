import { Card, CardContent, CardHeader } from '@mui/material'
import theme from '../../../styles/Theme'

export interface ModuleProps {
  title: string
  children?: JSX.Element
}

const Module = ({ title, children }: ModuleProps) => (
  <Card
    sx={{
      borderRadius: theme.border.radius.sm,
      margin: theme.spacing(1),
    }}
  >
    <CardHeader
      title={title}
      titleTypographyProps={{ variant: 'subtitle2' }}
      sx={{
        color: 'common.white',
        backgroundColor: theme.background.color.light,
        height: '16px',
        padding: theme.spacing(2),
      }}
    />
    <CardContent
      sx={{
        backgroundColor: theme.background.color.lightest,
        padding: theme.spacing(2),
        height: '16px',
      }}
    >
      {children}
    </CardContent>
  </Card>
)

export default Module
