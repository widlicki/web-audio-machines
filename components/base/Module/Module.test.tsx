import { render } from '@testing-library/react'
import Module from './Module'
import { DefaultModule } from './Module.stories'

describe('Module Stories', () => {
  new Map([['Default module', DefaultModule]]).forEach((key) => {
    it(`renders ${key}`, () => {
      const component = <Module title="test" />
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})
