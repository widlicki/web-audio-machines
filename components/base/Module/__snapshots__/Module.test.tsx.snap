// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Stories renders function () { [native code] } 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1w3wvhy-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-1aseecf-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle2 MuiCardHeader-title css-1sr7afs-MuiTypography-root"
        >
          test
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-198pza3-MuiCardContent-root"
    />
  </div>
</div>
`;
